### PR TITLE
Read language-server coordinates from smithy-build.json

### PR DESCRIPTION
**Pull request by daddykotex**
Try reading the coordinates for the language-server from
the smithy-build.json, if failed or not found, then it
fallback to the previous implementation.

### CODE REVIEW

**Comment by daddykotex** [comment]
I'm trying to add tests but this project uses jest ( because I changed it a while back) when the only examples I could find to run VS Code tests were using mocha. I can't install them side-by-side because typescript complains, and I was not able to adapt the examples to run with jest.

I'll try to go back to mocha or use figure out the jest thingy.

File: src/extension.ts [hunk 1]
```diff
@@ -14,6 +14,23 @@ import {
...
+
+interface SmithyBuild {
+  imports?: Array<string>;
+  mavenRepositories?: Array<string>; //deprecated
+  mavenDependencies?: Array<string>; //deprecated
+  maven?: {
+    dependencies?: Array<string>;
+    repositories?: Array<{ url: string }>;
+  };
+  languageServer?: MavenCoordinate;
```
Comment (kubukoz) [inline src/extension.ts:32]: Is it needed to read all of these fields? As far as I can see we're only interested in `languageServer`

Comment (daddykotex) [reply src/extension.ts:32]: I guess the content is read anyway, I'm just being explicit about the content, but yeah I could remove most have it and use `PartialSmithyBuild instead

Comment (kubukoz) [reply src/extension.ts:32]: yeah, it's gonna be the same either way as you're not really validating the file contents (`as` gets erased in the output javascript)

**Comment by daddykotex** [comment]
I'm playing a lot with this branch, will mark as draft in the meantime.

File: src/extension.ts (hunk 1 above)
Comment (daddykotex) [reply src/extension.ts:32]: if it were me, I'd keep it, I can add a not but I would keep it.

but if you really want it remove and I can, also wdyt @Baccata ?

**Review by Baccata — APPROVED** [review]

File: src/extension.ts (hunk 1 above)
Comment (Baccata) [reply src/extension.ts:32]: I think Jakub is right, but I'm merging anyway because time differences

**Pull request merged**